clang/lib/Sema/TreeTransform.h

```diff
@@ -13714,6 +13714,12 @@ TreeTransform<Derived>::TransformLambdaExpr(LambdaExpr *E) {
 
     // Capturing 'this' is trivial.
     if (C->capturesThis()) {
+      // We need ThisType when build capture in CheckCXXThisCapture.
```

Comment:
```suggestion
      // If this is a lambda that is part of a default member initialiser
      // and which we're instantiating outside the class that 'this' is 
      // supposed to refer to, adjust the type of 'this' accordingly.
      //
      // Otherwise, leave the type of 'this' as-is.
```
This still only describes *what* we’re doing below, not *why*; what I meant was that we should point out that the reason why we’re pushing a `this` scope here is because we may not be inside the class that `this` is supposed to refer to here; I’d suggest something like this perhaps.

Reply:
Ah, I see. I didn't realize to combine test case with the comment to have a better illustration. Sorry for my poor English and expression! I have pushed your suggestion.

Reply:
No problem—writing descriptive comments is hard; I had to think a while about how to phrase this, and I don’t think the way I phrased it is ideal either, but it’s hopefully good enough.

Reply:
Thanks for your patience!